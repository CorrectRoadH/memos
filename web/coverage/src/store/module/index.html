
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/store/module</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/store/module</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">24.63% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>170/690</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>9/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">9.25% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/54</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">24.63% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>170/690</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="dialog.ts"><a href="dialog.ts.html">dialog.ts</a></td>
	<td data-value="19.23" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 19%"></div><div class="cover-empty" style="width: 81%"></div></div>
	</td>
	<td data-value="19.23" class="pct low">19.23%</td>
	<td data-value="26" class="abs low">5/26</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="19.23" class="pct low">19.23%</td>
	<td data-value="26" class="abs low">5/26</td>
	</tr>

<tr>
	<td class="file medium" data-value="editor.ts"><a href="editor.ts.html">editor.ts</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="28" class="abs medium">16/28</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="28" class="abs medium">16/28</td>
	</tr>

<tr>
	<td class="file low" data-value="filter.ts"><a href="filter.ts.html">filter.ts</a></td>
	<td data-value="24.67" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 24%"></div><div class="cover-empty" style="width: 76%"></div></div>
	</td>
	<td data-value="24.67" class="pct low">24.67%</td>
	<td data-value="77" class="abs low">19/77</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="10" class="abs low">1/10</td>
	<td data-value="24.67" class="pct low">24.67%</td>
	<td data-value="77" class="abs low">19/77</td>
	</tr>

<tr>
	<td class="file low" data-value="global.ts"><a href="global.ts.html">global.ts</a></td>
	<td data-value="10.63" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 10%"></div><div class="cover-empty" style="width: 90%"></div></div>
	</td>
	<td data-value="10.63" class="pct low">10.63%</td>
	<td data-value="94" class="abs low">10/94</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="10.63" class="pct low">10.63%</td>
	<td data-value="94" class="abs low">10/94</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file low" data-value="layout.ts"><a href="layout.ts.html">layout.ts</a></td>
	<td data-value="22.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="18" class="abs low">4/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="18" class="abs low">4/18</td>
	</tr>

<tr>
	<td class="file low" data-value="memo.ts"><a href="memo.ts.html">memo.ts</a></td>
	<td data-value="25.4" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 25%"></div><div class="cover-empty" style="width: 75%"></div></div>
	</td>
	<td data-value="25.4" class="pct low">25.4%</td>
	<td data-value="122" class="abs low">31/122</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	<td data-value="25.4" class="pct low">25.4%</td>
	<td data-value="122" class="abs low">31/122</td>
	</tr>

<tr>
	<td class="file low" data-value="resource.ts"><a href="resource.ts.html">resource.ts</a></td>
	<td data-value="11.11" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 11%"></div><div class="cover-empty" style="width: 89%"></div></div>
	</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="90" class="abs low">10/90</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="90" class="abs low">10/90</td>
	</tr>

<tr>
	<td class="file low" data-value="shortcut.ts"><a href="shortcut.ts.html">shortcut.ts</a></td>
	<td data-value="14.28" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 14%"></div><div class="cover-empty" style="width: 86%"></div></div>
	</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="49" class="abs low">7/49</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="49" class="abs low">7/49</td>
	</tr>

<tr>
	<td class="file low" data-value="tag.ts"><a href="tag.ts.html">tag.ts</a></td>
	<td data-value="19.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 19%"></div><div class="cover-empty" style="width: 81%"></div></div>
	</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	</tr>

<tr>
	<td class="file low" data-value="user.ts"><a href="user.ts.html">user.ts</a></td>
	<td data-value="35.86" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.86" class="pct low">35.86%</td>
	<td data-value="145" class="abs low">52/145</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="35.86" class="pct low">35.86%</td>
	<td data-value="145" class="abs low">52/145</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-04-04T12:00:15.191Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    