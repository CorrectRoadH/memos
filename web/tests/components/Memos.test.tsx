import { render, screen } from "@testing-library/react";
import Memo from "../../src/components/Memo";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../src/store";
import renderer from "react-test-renderer";

describe("Memo", () => {
  it("render Memo", () => {
    const memo: Memo = {
      id: 123,
      creatorId: 1234,
      createdTs: 0,
      updatedTs: 0,
      rowStatus: "NORMAL" as RowStatus,
      content: "Hello!",
      visibility: "PUBLIC" as Visibility,
      pinned: false,
      creatorName: "test",
      resourceList: [],
    };
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Memo memo={memo} />
        </BrowserRouter>
      </Provider>
    );
    screen.getByText(/Hello!/i);
  });
  it("memo snapshot", () => {
    const memo: Memo = {
      id: 123,
      creatorId: 1234,
      createdTs: 0,
      updatedTs: 0,
      rowStatus: "NORMAL" as RowStatus,
      content: "Hello!",
      visibility: "PUBLIC" as Visibility,
      pinned: false,
      creatorName: "test",
      resourceList: [],
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Memo memo={memo} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
