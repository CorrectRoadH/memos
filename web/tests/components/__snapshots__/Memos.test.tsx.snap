// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Memo > memo snapshot 1`] = `
<div
  className="memo-wrapper memos-123 "
>
  <div
    className="memo-top-wrapper"
  >
    <div
      className="status-text-container"
    >
      <a
        className="time-text"
        href="/m/123"
        onClick={[Function]}
      >
        1970/01/01 08:00:00
      </a>
      <a
        className="name-text"
        href="/u/1234"
        onClick={[Function]}
      >
        @
        test
      </a>
    </div>
  </div>
  <div
    className="memo-content-wrapper "
  >
    <div
      className="memo-content-text "
      onClick={[Function]}
      onDoubleClick={[Function]}
    >
      <p>
        Hello!
      </p>
    </div>
  </div>
  <div
    className="resource-wrapper "
  />
</div>
`;
